### PR TITLE
Decouple navigation to inspector selection location from IntelliJ.

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Comment by jacob314** [comment]
Fixes https://github.com/flutter/flutter-intellij/issues/4929

File: src/io/flutter/inspector/InspectorService.java [hunk 1]
```diff
@@ -398,12 +398,11 @@ private void onVmServiceReceived(String streamId, Event event) {
     switch (streamId) {
       case VmService.DEBUG_STREAM_ID: {
         if (event.getKind() == EventKind.Inspect) {
-          // Make sure the WidgetInspector on the device switches to show the inspected object
-          // if the inspected object is a Widget or RenderObject.
+          // Assume the inspector in Flutter DevTools or the on device
+          // has already set the selection on the device so we don't have to.
+          // Having multiple clients set the selection risks race conditions
+          // where the selection ping-pongs back an forth.
```
Comment (helin24) [inline src/io/flutter/inspector/InspectorService.java:404]: typo - and

File: src/io/flutter/inspector/InspectorService.java [hunk 2]
```diff
@@ -398,12 +398,11 @@ private void onVmServiceReceived(String streamId, Event event) {
     switch (streamId) {
       case VmService.DEBUG_STREAM_ID: {
         if (event.getKind() == EventKind.Inspect) {
-          // Make sure the WidgetInspector on the device switches to show the inspected object
-          // if the inspected object is a Widget or RenderObject.
+          // Assume the inspector in Flutter DevTools or the on device
```
Comment (helin24) [inline src/io/flutter/inspector/InspectorService.java:401]: typo - on the

File: src/io/flutter/view/FlutterView.java [hunk 3]
```diff
@@ -115,6 +119,29 @@ public FlutterView(@NotNull Project project) {
...
+            sourcePosition.createNavigatable(project).navigate(true);
+          }
+          if (selection.isCreatedByLocalProject()) {
+            final XSourcePosition position = selection.getCreationLocation().getXSourcePosition();
+            if (position != null) {
+              position.createNavigatable(project).navigate(false);
+            }
+          }
+        }
+        System.out.println("XXX selection changed to :" + selection);
```
Comment (helin24) [inline src/io/flutter/view/FlutterView.java:142]: Is this for debugging?

**Review by helin24 — APPROVED** [review]

File: src/io/flutter/inspector/InspectorService.java (hunk 1 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java:404]: done

File: src/io/flutter/inspector/InspectorService.java (hunk 2 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java:401]: done

File: src/io/flutter/view/FlutterView.java (hunk 3 above)
Comment (jacob314) [reply src/io/flutter/view/FlutterView.java:142]: removed

**Comment by jacob314** [comment]
ptal

File: src/io/flutter/inspector/InspectorService.java [hunk 4]
```diff
@@ -398,12 +398,11 @@ private void onVmServiceReceived(String streamId, Event event) {
     switch (streamId) {
       case VmService.DEBUG_STREAM_ID: {
         if (event.getKind() == EventKind.Inspect) {
-          // Make sure the WidgetInspector on the device switches to show the inspected object
-          // if the inspected object is a Widget or RenderObject.
+          // Assume the inspector in Flutter DevTools or the on device widget
```
Comment (helin24) [inline src/io/flutter/inspector/InspectorService.java:401]: Assume the inspector in Flutter DevTools or **on the** device widget

File: src/io/flutter/inspector/InspectorService.java [hunk 5]
```diff
@@ -398,12 +398,11 @@ private void onVmServiceReceived(String streamId, Event event) {
...
+          // inspector has already set the selection on the device so we don't
+          // have to Having multiple clients set the selection risks race
```
Comment (helin24) [inline src/io/flutter/inspector/InspectorService.java:403]: add the period back after "have to."

**Review by helin24 — APPROVED** [review]

File: src/io/flutter/inspector/InspectorService.java (hunk 4 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java:401]: done

File: src/io/flutter/inspector/InspectorService.java (hunk 5 above)
Comment (jacob314) [reply src/io/flutter/inspector/InspectorService.java:403]: done

**Review by helin24 — APPROVED** [review]
Very legible!

**Pull request merged**